### PR TITLE
#577 Sitemap xml

### PR DESCRIPTION
**Pull request by jankaBehal**
Config of sitemap script that will run on each build and regenerate new sitemap, which is saved in public folder

Sitemap

### CODE REVIEW

File: next/next-sitemap.config.js [hunk 1]
```diff
@@ -0,0 +1,10 @@
+/** @type {import('next-sitemap').IConfig} */
+module.exports = {
+  siteUrl: process.env.SITE_URL || 'https://www.olo.sk',
+  generateRobotsTxt: true, // (optional)
```
Comment (radoslavzeman) [inline next/next-sitemap.config.js:4]: We already have `/api/robots` and relevant redirect in next config for robots file:

image

image

Obviously, it should be updated.

File: next/next-sitemap.config.js [hunk 2]
```diff
@@ -0,0 +1,10 @@
...
+  changefreq: 'weekly',
+  sitemapSize: 5000, // optional, it will split large sitemaps into multiple files
+  exclude: ['/404', '/form/*', '/articles/*', '/services/*', '/workshops/*'], // TODO include all slugs in translation
```
Comment (radoslavzeman) [inline next/next-sitemap.config.js:7]: `/documents/*`, `/career/*`, `/faq-categories/*`?

File: next/.env.example [hunk 3]
```diff
@@ -1,6 +1,7 @@
 # Copy this to .env.local
 NEXT_PUBLIC_DEPLOYMENT=dev
 NEXT_PUBLIC_NEXT_HOST=127.0.0.1:3000
+SITE_URL=http://www.olo.sk
```
Comment (radoslavzeman) [inline next/.env.example:4]: Is there `http` instead of `https` on purpose?

Comment (jankaBehal) [reply next/.env.example:4]: you are right this is a mistake

File: next/next-sitemap.config.js (hunk 2 above)
Comment (jankaBehal) [reply next/next-sitemap.config.js:7]: let discuss this

File: next/next-sitemap.config.js (hunk 1 above)
Comment (jankaBehal) [reply next/next-sitemap.config.js]: robots api is now updated to allow robots only in production

File: next/next-sitemap.config.js [hunk 4]
```diff
@@ -0,0 +1,10 @@
+/** @type {import('next-sitemap').IConfig} */
+module.exports = {
+  siteUrl: process.env.SITE_URL || 'https://www.olo.sk',
+  generateRobotsTxt: false, // (optional)
```
Comment (radoslavzeman) [inline next/next-sitemap.config.js:4]: Maybe add a reason in a comment please

**Pull request merged**